exports/scanresultscollector.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+import datetime
+import json
+import logging
+import os
+import random
+import string
+import tarfile
+import uuid
+from io import BytesIO
+from io import StringIO
+
+from google.cloud import storage
+
+
+class ScanResultsCollector:
+    """
+    Sends the result to a Google Cloud Storage bucket, for future use in DefectDojo
+    """
+
+    def __init__(self, keyfile, bucket_name, app_name, directory=None):
+        client = storage.Client.from_service_account_json(keyfile)
+        self.bucket = client.get_bucket(bucket_name)
+        self.directory = directory or f"RapiDAST-{app_name}"
+        self.app_name = app_name
+
+    def create_metadata(self, data):
+        """
+        Given a dictionary of key/values corresponding to Defectdojo's `import-scan` parameters,
+        return a Metadata dictionary
+        """
+        metadata = {
+            "scan_type": data["scan_type"],
+            "uuid": str(uuid.uuid1()),
+            "test_description": "This test was executed by RapiDAST",
+            "import_data": data,
+        }
+
+        return metadata
+
+    def export_scan(self, data, filename):
+        """
+        Send the scan to ScanResultsCollector
+
+        Params:
+        data: a dictionary of key/values corresponding to Defectdojo's `import-scan` parameters
+        filename: path to the file containing scan
+
+        """
+        if not data or not filename:
+            # missing data means nothing to do
+            logging.debug("Insufficient data for Defect Dojo")
```

Comment:
"Insufficient data" should be good enough.

Reply:
updated